### PR TITLE
Merge upstream changes

### PR DESCRIPTION
**Pull request by ClearlyClaire**
## Conflicts

- `.prettierignore`:
  Upstream added a line at the end, glitch-soc had extra entries at the end.
  Added upstream's new line before glitch-soc's.
- `Gemfile.lock`:
  Upstream updated dependencies while glitch-soc has an extra one (hcaptcha)
  Updated dependencies like upstream did.
- `app/controllers/api/v1/statuses_controller.rb`:
  Not a real conflict, upstream added a parameter (`allowed_mentions`) where glitch-soc already had an extra one (`content_type`).
  Added upstream's new parameter.
- `app/javascript/styles/fonts/roboto-mono.scss`:
  A lot of lines were changed upstream due to code style changes, and a lot of those lines had path changes to accomodate glitch-soc's theming system.
  Applied upstream's style changes.
- `app/javascript/styles/fonts/roboto.scss`:
  A lot of lines were changed upstream due to code style changes, and a lot of those lines had path changes to accomodate glitch-soc's theming system.
  Applied upstream's style changes.

## Ported changes (from earlier merges)

- bae17ebe5eab02879599ae8516cf6b3f6736b450

## Ported changes

- 4da5f77d929d6b83c134cae1eefbc8ba2db752f8
- eddfb33dfea6a17e71377d95498b557dd0194477
- db2c58d47ae0db8490a30cd3846f30e615c382b5
- 0c9d455ea5540e854c2d879cd1df015ea493f622

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -71,6 +71,9 @@ docker-compose.override.yml
 /app/javascript/mastodon/locales
 /config/locales
 
+# Ignore vendored CSS reset
+app/javascript/styles/mastodon/reset.scss
+
```
Comment (Plastikmensch) [inline .prettierignore:76]: Shouldn't this also include glitch-soc's `app/javascript/flavours/glitch/styles/reset.scss`?

Comment (ClearlyClaire) [reply .prettierignore:76]: You're right, I forgot that. I also forgot to port 0c9d455ea5540e854c2d879cd1df015ea493f622 as a whole.

**Review by Plastikmensch — CHANGES_REQUESTED** [review]
Overall this looks okay to me, just two minor things.

One general gripe I have with using prettier for style is that by default it enforces and wraps everything after 80 chars, which in some cases significantly decreases readability and skimmability imo. I'm not sure how to solve this though. Reverting those changes now would just cause them to be reintroduced with the next prettier run. Ignore comments and overrides are an option, but... I dunno.

Anyway, thanks for keeping glitch-soc alive and all the work you do

File: app/javascript/flavours/glitch/features/status/index.js [hunk 2]
```diff
@@ -5,7 +5,16 @@ import PropTypes from 'prop-types';
...
+import { fetchStatus,
+  muteStatus,
+  unmuteStatus,
+  deleteStatus,
+  editStatus,
+  hideStatus,
+  revealStatus,
+  translateStatus,
+  undoStatusTranslation,
+} from 'flavours/glitch/actions/statuses';
```
[the remark this answers is not in the document]
Comment (ClearlyClaire) [reply app/javascript/flavours/glitch/features/status/index.js:17]: This isn't on purpose, I'm not sure what happened here. Changing it to match upstream.

[the remark this answers is not in the document]
Comment (Plastikmensch) [reply app/javascript/flavours/glitch/features/status/index.js:17]: Thanks!
This one still differs for some reason

**Review by Plastikmensch — APPROVED** [review]

**Pull request merged**